MAVProxy/modules/mavproxy_soar.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+'''
+Soaring Module
+Ryan Friedman
+
+This module displays the estimated soaring thermals on the map.
+A circle is drawn with the estimated radius from ArduSoar's estimated thermal location.
+
+AP_FLAKE8_CLEAN
+'''
+
+from MAVProxy.modules.lib import mp_module, mp_util
+from MAVProxy.modules.mavproxy_map import mp_slipmap
+
+
+class soar(mp_module.MPModule):
+    def __init__(self, mpstate):
+        """Initialise module"""
+        super(soar, self).__init__(mpstate, "soar", "")
+
+    def mavlink_packet(self, m):
+        '''handle mavlink packets'''
+
+        if m.get_type() == 'NAMED_VALUE_FLOAT' and m.name.startswith("SOAR"):
+            if m.name == "SOAREKFX0":
+                self._strength = m.value
+            elif m.name == "SOAREKFX1":
+                self._radius = m.value
+            elif m.name == "SOAREKFX2":
+                self._x = m.value
+            elif m.name == "SOAREKFX3":
+                self._y = m.value
+            else:
+                raise NotImplementedError(m.name)
+
+            self.draw_thermal_estimate()
+
+    def draw_thermal_estimate(self):
+
+        if self._radius is None:
+            print("No radius")
+            return
+        if self._x is None:
+            print("No x")
+            return
+        if self._y is None:
+            print("No y")
+            return
+
+        home = self.module('wp').get_home()
```

Comment:
This module might not be loaded.

Reply:
.... also, are these thermals really home-relative or are they navigation-origin-relative?

Reply:
> .... also, are these thermals really home-relative or are they navigation-origin-relative?

Per `SoaringController::update_thermalling`:

```c++
    const AP_AHRS &_ahrs = AP::ahrs();
    if (!_ahrs.get_relative_position_NED_home(current_position)) {
        return;
    }
    ...
    
    // update the filter
    _ekf.update(_vario.reading, current_position.x, current_position.y, wind_drift.x, wind_drift.y);
```

Based on that, I take it as home-relative. Not great, but that's how it is today. Seems like moving home would break stuff. Anyways, yea, I currently draw them relative to home to match.